services/core/PlatformDriverAgent/platform_driver/interfaces/home_assistant.py

```diff
@@ -0,0 +1,308 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This material was prepared as an account of work sponsored by an agency of
+# the United States Government. Neither the United States Government nor the
+# United States Department of Energy, nor Battelle, nor any of their
+# employees, nor any jurisdiction or organization that has cooperated in the
+# development of these materials, makes any warranty, express or
+# implied, or assumes any legal liability or responsibility for the accuracy,
+# completeness, or usefulness or any information, apparatus, product,
+# software, or process disclosed, or represents that its use would not infringe
+# privately owned rights. Reference herein to any specific commercial product,
+# process, or service by trade name, trademark, manufacturer, or otherwise
+# does not necessarily constitute or imply its endorsement, recommendation, or
+# favoring by the United States Government or any agency thereof, or
+# Battelle Memorial Institute. The views and opinions of authors expressed
+# herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY operated by
+# BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+# }}}
+
+
+import random
+from math import pi
+import json
+import sys
+from platform_driver.interfaces import BaseInterface, BaseRegister, BasicRevert
+from volttron.platform.agent import utils #added this to pull from config store 
+from volttron.platform.vip.agent import Agent
+import logging
+import requests
+from requests import get
+from collections.abc import MutableMapping
+
+_log = logging.getLogger(__name__)
+type_mapping = {"string": str,
+                "int": int,
+                "integer": int,
+                "float": float,
+                "bool": bool,
+                "boolean": bool}
+
+# def flatten_dict(d: MutableMapping, parent_key: str = '', sep: str ='.') -> MutableMapping:
+#     items = []
+#     for k, v in d.items():
+#         new_key = parent_key + sep + k if parent_key else k
+#         if isinstance(v, MutableMapping):
+#             items.extend(flatten_dict(v, new_key, sep=sep).items())
+#         else:
+#             items.append((new_key, v))
+#     return dict(items)
+
+class HomeAssistantRegister(BaseRegister):
+    def __init__(self, read_only, pointName, units, reg_type, attributes,
+                 default_value=None, description=''):
+        super(HomeAssistantRegister, self).__init__("byte", read_only, pointName, units,
+                                           description='')
+        self.reg_type = reg_type
+        self.attributes = attributes
+
+        if default_value is None:
```

Comment:
Why 0-100 random as a default value?  If there is a good reason document it otherwise please have a sane SENTINAL value.